Src/BootCamp.Chapter1/ArrayOperations.cs

```diff
@@ -9,7 +9,31 @@ public static class ArrayOperations
         /// <param name="array">Input array in a random order.</param>
         public static void Sort(int[] array)
         {
-            // ToDo: implement.
+
+            if (array == null)
+            {
+                return;
+            }
+
+            var flag = true;
+            var temp = 0;
+            int numLength = array.Length; 
+
+            //sorting an array  
+            for (var i = 1; (i <= (numLength - 1)) && flag; i++)
+            {
+                flag = false;
+                for (var j = 0; j < (numLength - 1); j++)
+                {
+                    if (array[j + 1] < array[j])
+                    {
+                        temp = array[j];
+                        array[j] = array[j + 1];
+                        array[j + 1] = temp;
+                        flag = true;
+                    }
+                }
```

Comment:
if it wasn't sorted, break it

Reply:
but how do  now it is sorted or not. or can I assume is sorted. Then I need isSorted at all ?

Reply:
I think that your first version was not only correct but also better:

By using `int[] array = {5, 7, 3, 1, 2, 8, 6, 4, 9, 10, 20, 13, 15, 17, 19, 18, 11, 16, 12, 14};` my results are:
* With "flag":
  * Outer loop runs 19 times
  * Inner loop runs 361 times
* Without "flag":
  * Outer loop runs 8 times
  * Inner loop runs 152 times

So I suggest to use your old version but rename "flag" to "didSwap" or "wasSwapped" or whatever name feels good to you.

Reply:
Why ?

it seems to me that the without flag is better , it runs less times but still you asked me to put the old version back. so I do not understand why the old one is better ?

Reply:
Good question. I did a mistake when writing it down. I accidently switched with and without case.
However, the thing Kai asked for is that you break after the inner loop is completely done.
So you only need one if after your inner loop that checks if flag was used.
In my eyes this is identical to the solution you had. Maybe Kai can take another look and decide again. :)

Reply:
oke, then we have to wait till @Almantask  has time for it. I think it will be this evening.

Reply:
Guys, the flag was needed, but you didn't need to place it in the condition. What I meant was:
```
bool wasSwapped = false;
ffor (var i = 1; i < numLength; i++)
            {
                for (var j = 0; j < (numLength - 1); j++)
                {
                    if (array[j + 1] < array[j])
                    {
                        temp = array[j];
                        array[j] = array[j + 1];
                        array[j + 1] = temp;
                        wasSwapped = true;
                        break;  // array is sorted now so the loop can be leaved.       
                    }
                }
              if(!wasSwapped) break;
           }
```

Reply:
thanks for showing me what you meant.